wordpress-mobile/WordPress-Android
# Use `trackWithSiteDetails` to track failed media uploads in the editor

This a follow up to #19853 

The changes here track `editor_upload_media_*` events.

-----

## To Test:

- Turn on Airplane mode on the device
- Open the editor and add an image from device to a post
- Notice that the `editor_upload_media_failed` event in the app log includes the site info: `site_type`, `blog_id`, and `is_jetpack`
- Hit retry on the block and confirm the retry modal 
- - Notice that the `editor_upload_media_retried` event in the app log includes the site info: `site_type`, `blog_id`, and `is_jetpack`
-----

## Regression Notes

1. Potential unintended areas of impact

    Change in metric tracking 

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Verified that only new keys are added to the tracking event
    - Verified that the event ids are unchanged 

3. What automated tests I added (or what prevented me from doing so)

    No additional test coverage is needed.

-----

## PR Submission Checklist:


-----
## UI Changes Testing Checklist:
